src/Kliva/ViewModels/ActivityDetailViewModel.cs

```diff
@@ -194,10 +197,10 @@ public ActivityDetailViewModel(INavigationService navigationService, IStravaServ
 
             if (result == ContentDialogResult.Primary && !string.IsNullOrEmpty(dialog.Description))
             {
-                await _stravaService.PostComment(SelectedActivity.Id.ToString(), dialog.Description);
-                await LoadActivityDetails(SelectedActivity.Id.ToString());
-                ServiceLocator.Current.GetInstance<IMessenger>().Send<PivotMessage<ActivityPivots>>(new PivotMessage<ActivityPivots>(ActivityPivots.Comments, true, true));
+                await _stravaService.PostComment(activityId.ToString(), dialog.Description);
+                await LoadActivityDetails(activityId.ToString());
             }
+            ServiceLocator.Current.GetInstance<IMessenger>().Send<PivotMessage<ActivityPivots>>(new PivotMessage<ActivityPivots>(ActivityPivots.Comments, true, true), Tokens.ActivityPivotMessage);
```

Comment:
Shouldn't we only show the comments tab, if a comment was entered? So I would move this Send<> back into the if(){...}